File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (!nums.size()) return 0;
        set<int> list;
        
        for (int num: nums) list.insert(num);
        int counter = 1;
        int output = 1;

        for (int i: list){
            if (list.count(i + 1)){
                counter++;
            }
            else {
                if (counter > output) output = counter;
                counter = 1;
            }
        }
        return output;
        
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        from collections import Counter, defaultdict
        count = Counter(nums)
        dict_count = defaultdict(list)

        for i in count:
            dict_count[count[i]].append(i)
        
        output = []

        while (k > 0):
            max_element = max(dict_count.keys())
            if dict_count[max_element]:
                output.append(dict_count[max_element][-1])
                dict_count[max_element].pop()
                k -= 1
            else:
                dict_count.pop(max_element)
        
        return output


        
        
            